Continuous scrubbing support for videos

tar500 opened the pull request:
- Continuous scrubber updates support for MediaPlayerScrubbingNode
- Added stateful seeking support to MediaPlayer, implemented debouncing to reduce CPU overhead for unneeded frequent video frame decodings
- Implemented continuous scrubbing for video nodes 📹

tar500 committed:
Continuous scrubbing support for videos

trs-p commented:
[scrubbing bug.mp4.zip](https://github.com/peter-iakovlev/TelegramUI/files/3317955/scrubbing.bug.mp4.zip)
[scrubbing fixed.mp4.zip](https://github.com/peter-iakovlev/TelegramUI/files/3317954/scrubbing.fixed.mp4.zip)

tar500 committed:
Continuous seek optimizations.
Moved debouncing to MediaPlayerScrubbingNode so it will be applied to other video nodes as well.

tar500 committed:
Hide fetching/buffering status UI updates while seek is in progress

trs-p commented:
- Optimized ffmpeg player continuous seeking performance (now it's 2-3x times faster)
- Added support for Youtube videos seeking
- Fixed flashing buffering status indicator

trs-p commented:
Since project structure has changed a lot - I moved it to main repo https://github.com/TelegramMessenger/Telegram-iOS/pull/104
